### PR TITLE
[develop] We now accept array as condition parameters

### PR DESCRIPTION
**Pull request by annando**
With this change the following is now possible:
```
$uid = [1, 2, 3];
$fields = ['uid', 'name', 'nick'];
$r = dba::select('contact', $fields, ['uid' => $uid, 'self' => true]);
```
The condition will be expanded to ```WHERE `uid` IN (?, ?, ?) AND `self` = ?```

### CODE REVIEW

File: include/dba.php [hunk 1]
```diff
@@ -1235,16 +1225,32 @@ public static function count($table, array $condition = [])
 	 * @param array $condition
 	 * @return string
 	 */
-	private static function buildCondition(array &$condition = [])
+	private static function buildCondition(&$condition = [])
```
Comment (MrPetovan) [inline include/dba.php]: What is the case when a non-array would be passed to this function?

Comment (annando) [reply include/dba.php]: Then nothing is passed - for example a "select" without a condition.

**Comment by annando** [comment]
Please don't accept it at the moment. I'm experiencing some error in the log that I want to check at first.

File: include/dba.php (hunk 1 above)
Comment (MrPetovan) [reply include/dba.php]: Then there shouldn't be a call to `buildCondition` but the type hint should stay and the type check should be removed.

Comment (MrPetovan) [reply include/dba.php]: I just checked the two places `buildCondition` is called and in both places the default value is an array.

**Comment by annando** [comment]
Problem should be solved now.

**Comment by annando** [comment]
And I just realized that the ```each(...)``` that I'm using is deprecated in PHP 7.2. So I will look for a replacement. Additionally I want to check for a possible problem. So don't accept it at the moment. ;-)

File: include/dba.php (hunk 1 above)
Comment (MrPetovan) [inline include/dba.php]: Please restore the type hint and remove the type check inside the function.

**Comment by annando** [comment]
Now it should finally work.

**Pull request merged**